x-pack/legacy/plugins/security/server/routes/api/v1/authenticate.js

```diff
@@ -82,6 +83,30 @@ export function initAuthenticateApi({ authc: { login, logout }, config }, server
     }
   });
 
+  server.route({
+    method: 'GET',
+    path: '/api/security/v1/oidc/implicit',
+    config: { auth: false },
+    async handler(request, h) {
+      const legacyConfig = server.config();
+      const basePath = legacyConfig.get('server.basePath');
+
+      const nonce = await generateCSPNonce();
+      const cspRulesHeader = createCSPRuleString(legacyConfig.get('csp.rules'), nonce);
+      return h.response(`
+        <script nonce="${nonce}">
```

Comment:
__note:__ It seems it'd be better to include ` <!DOCTYPE html>` at least, will do.

Reply:
The w3c validator makes it seem like we should include ` <title>foo</title>` as well: https://validator.w3.org/nu/#textarea

Borrowing from https://mathiasbynens.be/notes/minimal-html who found https://github.com/whatwg/html/commit/5c7cf94f9c4add53d74842581f593ea7bb099a15, the `title` can be omitted when dealing with e-mails, but it sounds like we should be using it here.

Reply:
Huh, would never thought we may need title :) Let me read through this, thanks for sharing!

Reply:
Yeah, the spec agrees with you: https://html.spec.whatwg.org/multipage/semantics.html#the-title-element: `Neither tag is omissible.` :slightly_smiling_face: